### PR TITLE
master

### PR DESCRIPTION
**Pull request by brucou**
Updated README

### CODE REVIEW

File: .idea/vcs.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
```
Comment (achou11) [inline .idea/vcs.xml:6]: Assuming this file was added by accident?

File: README.md [hunk 2]
```diff
@@ -2,23 +2,95 @@
...
+> [...] In practice, several extensions are useful to enhance the expressive power and
+to describe more complex systems more efficiently
+
+*Source: [Hierarchical State Machines](https://link.springer.com/content/pdf/10.1007/3-540-44929-9_24.pdf), Mihalis Yannakakis, Bell Laboratories, 2000*
+  
+According to the extensions they incorporate, the following types of state machines are commonly distinguished:
+ 
+| Extension | Terminology | Characteristics |
+|:---|:---|:---|
+| - | basic finite state machine ([FSM]((https://en.wikipedia.org/wiki/Finite-state_machine)) | inputs, states, transitions |
```
Comment (achou11) [inline README.md]: > ([FSM]((

Additional parenthesis after the `]` messes up the rendering for the closing parenthesis at the end of this line :)

**Comment by achou11** [comment]
@brucou First of all, just wanted to say thank you for putting so much time and effort into this! Was honestly surprised that you were going to add this much content in 😅 I added a couple of small comments but overall it looks great at a glance!

I know that the topic has some deep roots in academia and some very technical knowledge surrounding it - is it possible to somehow make the explanations near the beginning a little more beginner-friendly? I think it's great, but I can imagine someone with no prior knowledge stumbling upon this and seeing a massive table and lots of relatively unfamiliar terminology being somewhat overwhelmed by it. Not really a big deal to be quite honest, but wanted to offer the question in case you did have any potential ideas regarding this. I really want to make this as approachable as possible for newcomers like myself. Just want to make sure I don't lose sight of that as we add content to this.

Really appreciate this PR! I look forward to going through some of the stuff you've added to educate myself :)

File: .idea/vcs.xml (hunk 1 above)
Comment (brucou) [reply .idea/vcs.xml:6]: absolutely. That file is not even on my computer anymore strangely. It is usually automatically added by webstorm my editor.

**Comment by brucou** [comment]
@achou11 Thanks for the comments! I thought you were pursuing an approach based on external links rather than generating a pedagogical resource on your own. The idea in any case in this PR is not to explain all the diferrent state machine extensions, but to refer to an explanation for more details. I will add some quick comments.

**Comment by achou11** [comment]
Makes sense! And thank you :)

Let me know when you feel it's ready and I'll merge it 👍

**Comment by brucou** [comment]
I think I am done for this PR. Please review one last time to see if you are comfortable with the changes. It will always be possible to modify stuff in other PRs anyways.

**Pull request merged**